runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java

```diff
@@ -591,7 +595,7 @@ public void flushState() {
               timerId,
               "",
               cleanupTime,
-              cleanupTime,
+              cleanupTime.minus(1L),
```

Comment:
I want to avoid the performance regression of setting a watermark hold if not needed.  Make the outputTimestamp be a parameter to this function, and pass it in only if the DoFn has an onWindowExpiration (SimpleParDoFn has the DoFnSignature so you should be able to tell there).

Reply:
done